libraries/provider_runit_service.rb

```diff
@@ -311,6 +304,11 @@ def default_logger_content
         #
         # Helper Resources
         #
+        def do_action(resource, action)
+          resource.run_action(action)
+          new_resource.updated_by_last_action(true) if resource.updated_by_last_action?
```

Comment:
No, you get automatic bubbling up from sub-resources in LWRP Providers because of the sub-resource collection:

https://github.com/chef/chef/blob/master/lib/chef/provider/lwrp_base.rb#L65-L67

In HWRPs like this you have to do it yourself.